flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/source/reader/split/PulsarPartitionSplitReaderBase.java

```diff
@@ -119,6 +123,12 @@ public RecordsWithSplitIds<PulsarMessage<OUT>> fetch() throws IOException {
                     collector.setMessage(message);
                     deserializationSchema.deserialize(message, collector);
 
+                    // invoke user callback
+                    if (userCallback != null) {
+                        callSafely(
+                                () -> userCallback.process(message, collector.getCollectedValue()));
```

Comment:
Do we need to deserialize it again in the callback function?

Reply:
no we don't. the contract for the callback is 

```
void process(Message<byte[]> rawMessage, T deserializedElement);
```

Reply:
Oh right.. miss the T there